test/test_format.py

```diff
@@ -0,0 +1,528 @@
+                '<mtr><mtd columnalign="center"><mglyph width="147px" height="175px" src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjEuMHB4IiBoZWlnaHQ9IjI1LjBweCIgeG1sbnM6c3ZnPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgICAgICAgICAgICAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgICAgICAgICAgICAgIHZlcnNpb249IjEuMSIKICAgICAgICAgICAgICAgIHZpZXdCb3g9IjYzLjAwMDAwMCA3NS4wMDAwMDAgMjEuMDAwMDAwIDI1LjAwMDAwMCI+CiAgICAgICAgICAgICAgICA8IS0tR3JhcGhpY3NFbGVtZW50cy0tPgo8dGV4dCB4PSI3My41IiB5PSI4Ny41IiBveD0iMCIgb3k9IjAiIGZvbnQtc2l6ZT0iMTBweCIgc3R5bGU9InRleHQtYW5jaG9yOmVuZDsgZG9taW5hbnQtYmFzZWxpbmU6aGFuZ2luZzsgc3Ryb2tlOiByZ2IoMC4wMDAwMDAlLCAwLjAwMDAwMCUsIDAuMDAwMDAwJSk7IHN0cm9rZS1vcGFjaXR5OiAxOyBmaWxsOiByZ2IoMC4wMDAwMDAlLCAwLjAwMDAwMCUsIDAuMDAwMDAwJSk7IGZpbGwtb3BhY2l0eTogMTsgY29sb3I6IHJnYigwLjAwMDAwMCUsIDAuMDAwMDAwJSwgMC4wMDAwMDAlKTsgb3BhY2l0eTogMS4wIj5hXmI8L3RleHQ+Cjwvc3ZnPgo="/></mtd></mtr>\n'
+                "</mtable>"
+            ),
+            None,
+        ),
+    ],
+)
+def test_makeboxes_outputform_mathml(
+    str_expr: str, str_expected: str, msg: str, message=""
+):
+    result = session.evaluate(str_expr)
+    format_result = result.format(session.evaluation, "System`OutputForm")
+    # Atoms are not still correctly processed as BoxConstruct
+    #    assert isinstance(format_result,  BoxConstruct)
+    if msg:
+        assert (
+            format_result.boxes_to_mathml(evaluation=session.evaluation) == str_expected
+        ), msg
+    else:
+        str_format = format_result.boxes_to_mathml(evaluation=session.evaluation)
+        print(f"<<{str_format}>>")
+        print(f"<<{str_expected}>>")
+        assert str_format == str_expected
+
+
+# Text StandardForm
+@pytest.mark.parametrize(
+    ("str_expr", "str_expected", "msg"),
+    [
+        ('"4"', "4", None),
+        ("4", "4", None),
+        ('"4.32"', "4.32", None),
+        ('"4.32313213213`2"', "4.32313213213`2", None),
+        ('"Hola!"', "Hola!", None),
+        ("a", "a", None),
+        ("Pi", "Pi", None),
+        ("a^4", "a^4", None),
+        # Boxerror
+        #        ("Subscript[a, 4]", "Subscript[a, 4]", None),
+        #        ("Subsuperscript[a, p, q]", "Subsuperscript[a, p, q]", None),
+        #        (
+        #            "Integrate[F[x],{x,a,g[b]}]",
+        #            "Subsuperscript[∫, a, g[b]]\u2062F[x]\u2062\uf74cx",
+        #            None,
+        #        ),
+        # This seems to be wrong...
+        ("a^(b/c)", "a^ ( b ) / ( c ) ", None),
+        ("1/(1+1/(1+1/a))", " ( 1 ) / ( 1+ ( 1 ) / ( 1+ ( 1 ) / ( a )  )  ) ", None),
+        # Boxerror
+        #        (
+        #            "Sqrt[1/(1+1/(1+1/a))]",
+        #            "Sqrt[ ( 1 ) / ( 1+ ( 1 ) / ( 1+ ( 1 ) / ( a )  )  ) ]",
+        #            None,
+        #        ),
+        ("Graphics[{}]", "-Graphics-", None),
+        # These tests requires ``evaluation`` as a parameter.
```

Comment:
These expressions require that an `evaluation` parameter be passed to format to produce the expression. This parameter should be optional.